lib/secretsManager.js

```diff
@@ -1,77 +1,169 @@
-let DUMMY_UUID_GEN = 0
-let DUMMY_STORE = []
-
-/**
- * Find all secrets, applying a selector if provided.
- * When provided, all where clause must be fulfilled (logical AND).
- * By default, returns all secrets
- * @async
- * @param {Object} where - hash of where clauses returned secrets must fulfill with
- * @returns {Array<Object>} list (may be empty) of matching secrets
- */
-export async function find(where = {}) {
-  let secrets = []
-
-  try {
-    const props = Object.keys(where)
-    secrets = DUMMY_STORE.filter(secret =>
-      props.every(prop => secret[prop] === where[prop])
+import { v4 as uuid } from 'uuid'
+import aesjs from 'aes-js'
+import * as SecureStore from 'expo-secure-store'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+const encryptionPassphraseStorageKey = 'encryptionPassphrase'
+const secretStorageKey = 'secrets'
+
+export default class SecretsManager {
+  constructor() {
+    this.encryptionPassphrase = null
+  }
+
+  async initialize() {
+    let getEncryptionKey = AsyncStorage.getItem
+    let setEncryptionKey = AsyncStorage.setItem
+
+    if (SecureStore.isAvailableAsync()) {
+      getEncryptionKey = SecureStore.getItemAsync
+      setEncryptionKey = SecureStore.setItemAsync
+    }
+
+    const encryptionPassphrase = await getEncryptionKey(
+      encryptionPassphraseStorageKey
+    )
+
+    if (encryptionPassphrase) {
+      this.encryptionPassphrase = encryptionPassphrase
+      return
+    }
+
+    this.encryptionPassphrase = this.generateEncryptionPassphrase()
+
+    await this.flush()
+
+    await setEncryptionKey(
+      encryptionPassphraseStorageKey,
+      this.encryptionPassphrase
     )
-  } catch (err) {
-    console.error('Failed to read secrets from data store', err)
   }
 
-  return secrets
-}
+  generateEncryptionPassphrase() {
+    return JSON.stringify(
+      Array.from(
+        crypto.getRandomValues(
+          Uint8Array.from(Array.from({ length: 32, value: 0 }))
+        )
+      )
+    )
+  }
+
+  async flush() {
```

Comment:
not sure about the naming. flush usually means persisting temporary changes. it looks like you're cleaning stuff up here instead

Reply:
🤦 I assumed the literal meaning of the word. I'll change it. Is `reset` fair enough as an alternative?

Reply:
Using reset for now!